rule/utils_test.go

```diff
@@ -30,3 +30,171 @@ func TestIsCallToExitFunction(t *testing.T) {
+				t.Errorf("hasCaps(%s) = %v; want %v", tt.varName, got, tt.expected)
+			}
+		})
+	}
+}
+
+func TestIsUpperConstFunction(t *testing.T) {
+	tests := []struct {
+		varName  string
+		expected bool
+	}{
+		{"SOME_CONST_2", true},
+		{"__FOO", false},
+		{"__", false},
+		{"X509B", true},
+		{"FOO", true},
+		{"1FOO", false},
+		{"_FOO123_BAR456", true},
+		{"A1_B2_C3", true},
+		{"A1_b2", false},
+		{"FOO_", false},
+		{"foo", false},
+		{"_", false},
+		{"", false},
+		{"FOOBAR", true},
+		{"FO", true},
+		{"F_O", true},
+		{"FOO123", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.varName, func(t *testing.T) {
+			if got := isUpperCaseConst(tt.varName); got != tt.expected {
+				t.Errorf("isUpperCaseConst(%s) = %v; want %v", tt.varName, got, tt.expected)
+			}
+		})
+	}
+}
+
+func TestIsUpperUnderScoreFunction(t *testing.T) {
+	tests := []struct {
+		varName  string
+		expected bool
+	}{
+		{"_", false},
+		{"", false},
+		{"empty string", false},
+		{"_404_404", true},
+		{"FOO_BAR", true},
+		{"FOOBAR", false},
+		{"FO", false},
+		{"F_O", false},
+		{"_FOOBAR", true},
+		{"FOOBAR_", true},
+		{"FOO123", false},
+		{"FOO_123", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.varName, func(t *testing.T) {
+			if got := isUpperUnderscore(tt.varName); got != tt.expected {
+				t.Errorf("isUpperUnderScore(%s) = %v; want %v", tt.varName, got, tt.expected)
+			}
+		})
+	}
+}
+
+func TestIsDigitFunction(t *testing.T) {
+	tests := []struct {
+		input    rune
+		expected bool
+	}{
+		{'0', true},
+		{'1', true},
+		{'2', true},
+		{'3', true},
+		{'4', true},
+		{'5', true},
+		{'6', true},
+		{'7', true},
+		{'8', true},
+		{'9', true},
+		{'a', false},
+		{'Z', false},
+		{' ', false},
+		{'!', false},
+		{'🙂', false}, // Emoji to test unicode
+		{'٠', false}, // Arabic-Indic 0
+		{'١', false}, // Arabic-Indic 1
+		{'२', false}, // Devanagari 2
+		{'৩', false}, // Bengali 3
+		{'४', false}, // Devanagari 4
+		{'௫', false}, // Tamil 5
+		{'๖', false}, // Thai 6
+		{'৭', false}, // Bengali 7
+		{'८', false}, // Devanagari 8
+		{'९', false}, // Devanagari 9
+	}
+
+	for _, tt := range tests {
+		result := isDigit(tt.input)
+		if result != tt.expected {
+			t.Errorf("isDigit(%q) = %v; want %v", tt.input, result, tt.expected)
+		}
+	}
+}
+
+func TestIsUpperFunction(t *testing.T) {
+	tests := []struct {
+		input    rune
+		expected bool
+	}{
+		{'0', false},
+		{'5', false},
+		{'a', false},
+		{'A', true},
+		{'Ą', false},
+		{'Ć', false},
+		{'B', true},
+		{'C', true},
+		{'Z', true},
+		{' ', false},
+		{'!', false},
+		{'🙂', false}, // Emoji to test unicode
+		{'你', false},
+		{'日', false},
+		{'本', false},
+		{'語', false},
+		{'韓', false},
+		{'中', false},
+		{'文', false},
+		{'あ', false},
+		{'ア', false},
+		{'한', false},
+	}
+
+	for _, tt := range tests {
+		result := isUpper(tt.input)
+		if result != tt.expected {
+			t.Errorf("isUpper(%q) = %v; want %v", tt.input, result, tt.expected)
+		}
+	}
+}
```

Comment:
Here is what I suggest

```suggestion
func TestIsUpperFunction(t *testing.T) {
	t.Run("non letter", func(t *testing.T) {
	tests := []rune
		'0',
		'5'
		' ',
		'_',
		'!',
		'🙂', // Emoji to test unicode
	}

	for _, r := range tests {
		result := isUpper(r)
		if result {
			t.Errorf("isUpper(%q) = %v; want false", r, result)
		}
	}
})

t.Run("non ASCII letter", func(t *testing.T) {
	tests := []rune
		'Ą',
		'Ć',
		'你',
		'日',
		'本',
		'語',
		'韓',
		'中',
		'文',
		'あ',
		'ア',
		'한',
	}

	for _, r := range tests {
		result := isUpper(r)
		if result {
			t.Errorf("isUpper(%q) = %v; want false", r, result)
		}
	}
})

t.Run("lowercase ASCII letter", func(t *testing.T) {
	tests := []rune
		'a',
		'b',
	}

	for _, r := range tests {
		result := isUpper(r)
		if result {
			t.Errorf("isUpper(%q) = %v; want false", r, result)
		}
	}
})

t.Run("uppercase ASCII letter", func(t *testing.T) {
	tests := []rune
		'A',
		'B',
		'C', 
		'Z', 
	}

	for _, r := range tests {
		result := isUpper(r)
		if !result {
			t.Errorf("isUpper(%q) = %v; want true", r, result)
		}
	}
})
}
```